src/sentry/auth/provider.py

```diff
@@ -15,9 +19,8 @@ class MigratingIdentityId(namedtuple("MigratingIdentityId", ["id", "legacy_id"])
 
     __slots__ = ()
 
-    def __unicode__(self):
-        # Default to id when coercing for query lookup
-        return self.id
+    def __str__(self):
+        return force_text(self.id)
```

Comment:
Isn’t force_text spitting back unicode? So wouldn’t this be wrong for py2?

Reply:
Yeah, that's what `python_2_unicode_compatible` fixes.

https://consideratecode.com/2017/08/25/what-does-python_2_unicode_compatible-do/

Reply:
Welp.

> When using the decorator, __str__ should always return a unicode string. Therefor we cast the string to text_type.

That’s unintuitive.